quickjs.c

```diff
@@ -72,37 +72,49 @@
 #define CONFIG_ATOMICS
 #endif
 
-/* dump object free */
-//#define DUMP_FREE
-//#define DUMP_CLOSURE
-/* dump the bytecode of the compiled functions: combination of bits
-   1: dump pass 3 final byte code
-   2: dump pass 2 code
-   4: dump pass 1 code
-   8: dump stdlib functions
-  16: dump bytecode in hex
-  32: dump line number table
-  64: dump compute_stack_size
-  128: dump executed bytecode
- */
-//#define DUMP_BYTECODE  (1)
-/* dump the occurence of the automatic GC */
-//#define DUMP_GC
-/* dump objects freed by the garbage collector */
-//#define DUMP_GC_FREE
-/* dump objects leaking when freeing the runtime */
-//#define DUMP_LEAKS  1
-/* dump memory usage before running the garbage collector */
-//#define DUMP_MEM
+/* dump the bytecode of the compiled functions */
+//#define DUMP_BYTECODE  (1)   /* uncomment this line and specify bit combination */
+//#define DUMP_BYTECODE_FLAG_PASS3   0x01   /* 1: dump pass 3 final byte code */
+//#define DUMP_BYTECODE_FLAG_PASS2   0x02   /* 2: dump pass 2 code */
+//#define DUMP_BYTECODE_FLAG_PASS1   0x04   /* 4: dump pass 1 code */
+//#define DUMP_BYTECODE_FLAG_STDLIB  0x08   /* 8: dump stdlib functions */
+//#define DUMP_BYTECODE_FLAG_HEX     0x10   /* 16: dump bytecode in hex */
+//#define DUMP_BYTECODE_FLAG_PC2LINE 0x20   /* 32: dump line number table */
+//#define DUMP_BYTECODE_FLAG_STACK   0x40   /* 64: dump compute_stack_size */
+//#define DUMP_BYTECODE_FLAG_STEP    0x80   /* 128: dump executed bytecode */
+
+// dump options selected dynamically with -d<hex mask>.
```

Comment:
This is a bit weird. If we have an API to enable dumping but one doesn't uncomment the defined it doesn't really work, which is somewhat unexpected.

Perhaps we could have some if-def for all of them at once, and then you can select which one you want with the flags? We could set that one on Debug builds for example. Or would the overhead be too much?

Reply:
Actually, these defines are not used ATM for `DUMP_BYTECODE`, which is always dynamically selected. The `DUMP_XXX_FLAG` could be defined all the time or if not already defined using ifdefs if the `DUMP_XXX` is defined.  Work in progress I guess. 
I shall post a commit to make the output unconditional if the `DUMP_XXX_FLAG` is not defined already to simplify debug builds but I want to have a simple way to make the output conditional for interactive testing.

Reply:
Conversation resolved with more elegant approach in latest commit